Updates to UI style and functionality of file sidebar and properties panel

allison-truhlar opened the pull request:
@krokicki @neomorphic 
As of Tuesday 3/25 afternoon, I have this version of the extension installed on [fileglancer-dev.int.janelia.org](https://fileglancer-dev.int.janelia.org) if you want to view it.

This PR includes changes from several tickets:

[File browser UI](https://app.clickup.com/t/86a72z6ay):

- Adds filter functionality for the file share paths in the left sidebar
- Hide dots files and folders by default; adds a button at the top of the file list to toggle their visibility

[Collapsible properties panel](https://app.clickup.com/t/86a76k41k):
- Changes the component from a modal to a panel that slides in from the right side of the screen, squeezing the file list to the left while leaving it accessible for browsing
- Right clicking on a file opens a context menu. Right now, the only link in the context menu is to open the properties panel.
- Adds a button at the top of the file list to open the properties panel.
- Inside the properties panel, changes the styling of the permissions icons from green and red with a square outline to all black with no outline, to prevent them from looking clickable
- Inside the properties panel, changes the text of the file conversion button from "submit ticket" to "submit"

[Light and dark mode colors for UI](https://app.clickup.com/t/86a7c5zyd):
- Adds a button in the navbar to toggle light and dark mode
- Adds some custom theming colors (HHMI blue/green and a purple)
- Updates styles to use the `mtconfig` (Material Tailwind) custom color names, making for better light-dark transitions

I also took a first go at an icon, following the suggestion from our last meeting of just sticking the eye icon inside a file icon.

krokicki commented:
Looks great and works for me. The alternating colors in the zone rows are a little confusing because the color is very close to the mouseover highlight color: 
![Screenshot 2025-03-25 at 3 00 01 PM](https://github.com/user-attachments/assets/cbf3ac24-14d5-45e8-a176-f45bb967ae12)
Also, when the list gets long it can't be scrolled. It would be nice if you could scroll it independently of the right side, like in Starfish.

allison-truhlar commented:
Thanks, @krokicki. I think I made the two changes you requested (scrollbar for the zones and more color differentiation between light zone rows and highlighted zone rows), but I right now I only know how see my home directory in JupyterHub, so I'm not certain. Could you check on [fileglancer-dev.int.janelia.org](https://fileglancer-dev.int.janelia.org/) when you have a chance?

I also fixed a problem I noticed with the check boxes on the file list - although the checks were appearing correctly, only the most recently checked file remained highlighted. And I refactored the theme logic out of the navbar component into a hook, to follow the format I've been trying to use in other components - keep logic in hooks, layout/styling in the component.
